pkg/workloads/cortex/tf_api/api.py

```diff
@@ -116,199 +70,45 @@ def after_request(response):
-
-def parse_response_proto(response_proto):
-    results_dict = json_format.MessageToDict(response_proto)
-    outputs = results_dict["outputs"]
-    outputs_simplified = {}
-    for key in outputs:
-        value_key = DTYPE_TO_VALUE_KEY[outputs[key]["dtype"]]
-        outputs_simplified[key] = outputs[key][value_key]
-    return outputs_simplified
-
-
-def run_predict(payload, debug=False):
-    ctx = local_cache["ctx"]
-    api = local_cache["api"]
-    request_handler = local_cache.get("request_handler")
-
-    prepared_payload = payload
-
-    debug_obj("payload", payload, debug)
-    if request_handler is not None and util.has_function(request_handler, "pre_inference"):
-        try:
-            prepared_payload = request_handler.pre_inference(
-                payload,
-                local_cache["model_metadata"]["signatureDef"],
-                api["tensorflow"]["metadata"],
-            )
-            debug_obj("pre_inference", prepared_payload, debug)
-        except Exception as e:
-            raise UserRuntimeException(
-                api["tensorflow"]["request_handler"], "pre_inference request handler", str(e)
-            ) from e
-
-    validate_payload(prepared_payload)
-
-    prediction_request = create_prediction_request(prepared_payload)
-    response_proto = local_cache["stub"].Predict(prediction_request, timeout=300.0)
-    result = parse_response_proto(response_proto)
-    debug_obj("inference", result, debug)
-
-    if request_handler is not None and util.has_function(request_handler, "post_inference"):
-        try:
-            result = request_handler.post_inference(
-                result, local_cache["model_metadata"]["signatureDef"], api["tensorflow"]["metadata"]
-            )
-            debug_obj("post_inference", result, debug)
-        except Exception as e:
-            raise UserRuntimeException(
-                api["tensorflow"]["request_handler"], "post_inference request handler", str(e)
-            ) from e
-
-    return result
-
-
-def validate_payload(payload):
-    signature = local_cache["parsed_signature"]
-    for input_name, _ in signature.items():
-        if input_name not in payload:
-            raise UserException('missing key "{}"'.format(input_name))
-
-
-def prediction_failed(reason):
-    message = "prediction failed: {}".format(reason)
-    cx_logger().error(message)
-    return message, status.HTTP_406_NOT_ACCEPTABLE
-
-
-@app.route("/healthz", methods=["GET"])
-def health():
-    return jsonify({"ok": True})
-
-
 @app.route("/predict", methods=["POST"])
 def predict():
     debug = request.args.get("debug", "false").lower() == "true"
 
     try:
         payload = request.get_json()
-    except Exception as e:
+    except:
         return "malformed json", status.HTTP_400_BAD_REQUEST
 
-    ctx = local_cache["ctx"]
     api = local_cache["api"]
-
-    response = {}
+    predictor = local_cache["predictor"]
 
     try:
-        result = run_predict(payload, debug)
+        try:
```

Comment:
should we remove one layer of try/except here, or is there a reason to have them both?